Userland/Libraries/LibWeb/Layout/FlexFormattingContext.cpp

```diff
@@ -1057,7 +1073,21 @@ void FlexFormattingContext::determine_hypothetical_cross_size_of_item(FlexItem&
 
     // If we have a definite cross size, this is easy! No need to perform layout, we can just use it as-is.
     if (has_definite_cross_size(item.box)) {
-        item.hypothetical_cross_size = css_clamp(resolved_definite_cross_size(item), clamp_min, clamp_max);
+        // To avoid subtracting padding and border twice for `box-sizing: border-box` only min and max clamp should happen on a second pass
+        if (resolve_percentage_min_max_sizes) {
+            item.hypothetical_cross_size = css_clamp(item.hypothetical_cross_size, clamp_min, clamp_max);
+            return;
+        }
+
+        float cross_size = [&]() {
+            if (item.box.computed_values().box_sizing() == CSS::BoxSizing::BorderBox) {
```

Comment:
probably it would be good to also use `get_pixel_width` and `get_pixel_height` here.